google/detectors/credentials/generic_weak_credential_detector/src/main/java/com/google/tsunami/plugins/detectors/credentials/genericweakcredentialdetector/testers/rabbitmq/RabbitMQCredentialTester.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.testers.rabbitmq;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.provider.TestCredential;
+import com.google.tsunami.plugins.detectors.credentials.genericweakcredentialdetector.tester.CredentialTester;
+import com.google.tsunami.proto.NetworkService;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+import javax.inject.Inject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+/** Credential tester for RabbitMQ Management Portal. */
+public final class RabbitMQCredentialTester extends CredentialTester {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private final HttpClient httpClient;
+
+  private static final String RABBITMQ_SERVICE = "rabbitmq";
+  private static final String RABBITMQ_PAGE_TITLE = "RabbitMQ Management";
+  private static final String RABBITMQ_SERVER_HEADER = "Cowboy";
+  private static final String RABBITMQ_WWW_HEADER = "Basic realm=\"RabbitMQ Management\"";
+
+  @Inject
+  RabbitMQCredentialTester(HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public String name() {
+    return "RabbitMQCredentialTester";
+  }
+
+  @Override
+  public String description() {
+    return "RabbitMQ credential tester.";
+  }
+
+  @Override
+  public boolean batched() {
+    return false;
+  }
+
+  private static String buildTargetUrl(NetworkService networkService, String path) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    targetUrlBuilder.append(path);
+    return targetUrlBuilder.toString();
+  }
+
+  // Checks if the response body contains the title element of rabbitmq management page.
+  // Custom fingerprint phase.
+  private static boolean bodyContainsRabbitMQElements(String responseBody) {
```

Comment:
Minor style issue (Unfortunately I cannot fix it myself): Methods and non-static variables should be named in lowerCamelCase, with acronyms treated as words 

```diff
-  private static boolean bodyContainsRabbitMQElements(String responseBody) {
+ private static boolean bodyContainsRabbitMqElements(String responseBody) {

{...}

- .map(RabbitMQCredentialTester::bodyContainsRabbitMQElements)
+ .map(RabbitMQCredentialTester::bodyContainsRabbitMqElements)
```